src/protocol_auth.c

```diff
@@ -409,37 +311,32 @@ bool id_h(meshlink_handle_t *mesh, connection_t *c, const char *request) {
 
 	/* Check if version matches */
 
-	if(c->protocol_major != mesh->self->connection->protocol_major) {
+	if(c->protocol_major != PROT_MAJOR) {
 		logger(mesh, MESHLINK_ERROR, "Peer %s uses incompatible version %d.%d",
 		       c->name, c->protocol_major, c->protocol_minor);
 		return false;
 	}
 
-	if(!c->config_tree) {
-		init_configuration(&c->config_tree);
+	/* Check if we know this node */
 
-		if(!read_host_config(mesh, c->config_tree, c->name)) {
-			logger(mesh, MESHLINK_ERROR, "Peer %s has unknown identity", c->name);
-			return false;
-		}
-	}
+	node_t *n = lookup_node(mesh, c->name);
 
-	bool blacklisted = false;
-	get_config_bool(lookup_config(c->config_tree, "blacklisted"), &blacklisted);
+	if(!n) {
+		logger(mesh, MESHLINK_ERROR, "Peer %s has unknown identity", c->name);
+		return false;
+	}
 
-	if(blacklisted) {
+	if(n->status.blacklisted) {
 		logger(mesh, MESHLINK_EPEER, "Peer %s is blacklisted", c->name);
 		return false;
 	}
 
-	read_ecdsa_public_key(mesh, c);
+	node_read_public_key(mesh, n);
```

Comment:
Should the node handle has to be explicitly request for it's public key? While reading configuration don't we store public key it's handle?

Reply:
No, we only read part of the configuration by default (device class, blacklist status and submesh information). These don't require extra memory to be allocated. Only when we actually want to make a connection (either a meta-connection or UDP) to another node do we read more information from its config file.